app/form_models/offender_sar_case_form.rb

```diff
@@ -87,6 +88,11 @@ def subject_type
 
   private
 
+  def check_valid_dates_for_step(step)
+    @case.validate_date_of_birth if step == "subject-details"
+    @case.validate_received_date if step == "date-received"
+  end
```

Comment:
Is it time to replace the string `"subject-details"` with a named Constant?

Reply:
Can you give me an example of how that would look above?

Reply:
Perhaps in the form object defining the step names as constant strings or similar eg:

```
STEPS = %w(
  subject-details
  other
  step
  names
)
```

That allows easy validation of step names and consistency

E.g.

`if STEPS.includes?(step)`

Reply:
Got that already :) 

![image](https://user-images.githubusercontent.com/1161161/61123998-b5e4c400-a49d-11e9-8e42-5fc7af433851.png)

(incidentally, in my opinion the answer to that question in the comment is "no"

Reply:
> That allows easy validation of step names and consistency

Not really sure how that would relates to the specific lines you commented on, though.